tests/integration_tests/test_model_training_options.py

```diff
@@ -0,0 +1,204 @@
+@pytest.mark.parametrize('skip_save_model', [False, True])
+def test_model_progress_save(
+        skip_save_progress,
+        skip_save_model,
+        generated_data,
+        tmp_path
+):
+
+    input_features, output_features = get_feature_definitions()
+
+    model_definition = {
+        'input_features': input_features,
+        'output_features': output_features,
+        'combiner': {'type': 'concat', 'fc_size': 14},
+        'training': {'epochs': 10}
+    }
+
+    # create sub-directory to store results
+    results_dir = tmp_path / 'results'
+    results_dir.mkdir()
+
+    # run experiment
+    exp_dir_name = full_experiment(
+        data_train_df=generated_data.train_df,
+        data_validation_df=generated_data.validation_df,
+        data_test_df=generated_data.test_df,
+        output_directory=str(results_dir),
+        model_definition=model_definition,
+        skip_save_processed_input=True,
+        skip_save_progress=skip_save_progress,
+        skip_save_unprocessed_output=True,
+        skip_save_model=skip_save_model,
+        skip_save_log=True
+    )
+
+    #========== Check for required result data sets =============
+    if skip_save_model:
+        assert not os.path.isfile(
+            os.path.join(exp_dir_name, 'model', 'model_weights.index')
+        )
+    else:
+        assert os.path.isfile(
+            os.path.join(exp_dir_name, 'model', 'model_weights.index')
+        )
+
+    if skip_save_progress:
+        assert not os.path.isfile(
+            os.path.join(exp_dir_name, 'model', 'model_weights_progress.index')
+        )
+    else:
+        assert os.path.isfile(
+            os.path.join(exp_dir_name, 'model', 'model_weights_progress.index')
+        )
+
+
+# work-in-progress
+def test_model_save_resume(generated_data, tmp_path):
+
+    input_features, output_features = get_feature_definitions()
+    model_definition = {
+        'input_features': input_features,
+        'output_features': output_features,
+        'combiner': {'type': 'concat', 'fc_size': 14},
+        'training': {'epochs': 30, 'early_stop': 5}
+    }
+
+    # create sub-directory to store results
+    results_dir = tmp_path / 'results'
+    results_dir.mkdir()
+
+    exp_dir_name = full_experiment(
+        model_definition,
+        data_train_df=generated_data.train_df,
+        data_validation_df=generated_data.validation_df,
+        data_test_df=generated_data.test_df,
+        output_directory=results_dir
+    )
+
+    full_experiment(
+        model_definition,
+        data_train_df=generated_data.train_df,
+        model_resume_path=exp_dir_name
+    )
+
+    test_fp = os.path.join(str(tmp_path), 'data_to_predict.csv')
+    generated_data.test_df.to_csv(
+        test_fp,
+        index=False
+    )
+
+    full_predict(os.path.join(exp_dir_name, 'model'), data_csv=test_fp)
+
+    y_pred = np.load(os.path.join(exp_dir_name, 'y_predictions.npy'))
+
+    mse = mean_squared_error(y_pred, generated_data.test_df['y'])
```

Comment:
what we can do here is that after the first dull experiment we load the numpy predictions, and after the second experiment with resume we load the numpy predictions and then we assert that they are the same with `np.isclose(first_preds, second_preds)`

Reply:
I made the recommended change in the test.  Here is the commit https://github.com/uber/ludwig/pull/739/commits/5a33a452d97e06b73a46ef36172d1add679e63d2 

I think either restore may not be saving the weights or the resume is not loading the weights correctly.  The last epoch on the first `full_experiment` looks like this
```
Epoch 28
Training: 100%|██████████| 3/3 [00:00<00:00, 65.62it/s]
Evaluation train: 100%|██████████| 3/3 [00:00<00:00, 110.61it/s]
Evaluation vali : 100%|██████████| 1/1 [00:00<00:00, 117.43it/s]
Evaluation test : 100%|██████████| 1/1 [00:00<00:00, 116.25it/s]
Took 0.1004s
Took 0.1004s
╒═══════╤═════════╤═════════╤══════════════════════╤═══════════════════════╤════════╕
│ y     │    loss │   error │   mean_squared_error │   mean_absolute_error │     r2 │
╞═══════╪═════════╪═════════╪══════════════════════╪═══════════════════════╪════════╡
│ train │ 20.6874 │ -3.7530 │              20.6874 │                3.8724 │ 0.9998 │
├───────┼─────────┼─────────┼──────────────────────┼───────────────────────┼────────┤
│ vali  │ 24.5326 │ -4.2580 │              24.5326 │                4.2998 │ 0.9997 │
├───────┼─────────┼─────────┼──────────────────────┼───────────────────────┼────────┤
│ test  │ 25.1545 │ -4.3416 │              25.1545 │                4.3740 │ 0.9997 │
╘═══════╧═════════╧═════════╧══════════════════════╧═══════════════════════╧════════╛
```

On the second `full_experiment` with the resume, the first epoch report is epoch 28, which I think makes sense but the values don't look correct
```
Resuming training of model: /tmp/pytest-of-root/pytest-0/test_model_save_resume0/results/experiment_run/model
Resuming training of model: /tmp/pytest-of-root/pytest-0/test_model_save_resume0/results/experiment_run/model

Epoch 28

Epoch 28
Training: 100%|██████████| 3/3 [00:00<00:00, 40.29it/s]
Evaluation train: 100%|██████████| 3/3 [00:00<00:00, 110.75it/s]
Evaluation vali : 100%|██████████| 1/1 [00:00<00:00, 115.91it/s]
Evaluation test : 100%|██████████| 1/1 [00:00<00:00, 118.83it/s]
Took 0.1300s
Took 0.1300s
╒═══════╤══════════╤═════════════╤══════════════════════╤═══════════════════════╤═════════╕
│ y     │     loss │       error │   mean_squared_error │   mean_absolute_error │      r2 │
╞═══════╪══════════╪═════════════╪══════════════════════╪═══════════════════════╪═════════╡
│ train │ 458.7538 │ 287591.8750 │             460.9559 │           287591.8438 │ -2.3807 │
├───────┼──────────┼─────────────┼──────────────────────┼───────────────────────┼─────────┤
│ vali  │ 514.6791 │ 339206.7188 │             514.6791 │           339206.7188 │ -3.0783 │
├───────┼──────────┼─────────────┼──────────────────────┼───────────────────────┼─────────┤
│ test  │ 495.4767 │ 311607.8438 │             495.4767 │           311607.8438 │ -3.2109 │
╘═══════╧══════════╧═════════════╧══════════════════════╧═══════════════════════╧═════════╛
```